backend/src/auth/credentials.ts

```diff
@@ -76,6 +76,19 @@ export const checkPassword = (password: string, existingHash: string) => {
   return diff === 0;
 };
 
+export const verifyAdminToken = (token: string) => {
+  try {
+    const { exp } = jwt.verify(token, config.JWT_SECRET_TOKEN) as { exp };
+    const { isAdmin } = jwt.decode(token, config.JWT_SECRET_TOKEN);
```

Comment:
You should be able to fold this into the first verify call, as we're now decoding the token twice.